rxandroidble/src/main/java/com/polidea/rxandroidble2/RxBleConnection.java

```diff
@@ -584,17 +584,28 @@ Completable requestConnectionPriority(
      */
     int getMtu();
 
+    /**
+     * Performs GATT read PHY operation.
+     *
+     * @return Observable emitting the read Tx and Rx values.
+     */
     @RequiresApi(26 /* Build.VERSION_CODES.O */)
     Single<PhyPair> readPhy();
 
     /**
      * Performs set preferred PHY request.
      *
+     * @param txPhy Sets the preferred transmitter (Tx) PHY.
+     * @param rxPhy Sets the preferred receiver (Rx) PHY.
+     * @param phyOptions Sets the preferred coding to use when transmitting on the LE Coded PHY.
      * @return Observable emitting whether the GATT operation succeeded or not.
      * @throws BleGattException in case of GATT operation error with {@link BleGattOperationType#PHY_UPDATE} type.
+     * @implNote RxBlePhy.PHY_UNKNOWN is used for the onPhyRead and onPhyUpdate callbacks in cases where the GATT operation
+     *           was not successful. Using RxBlePhy.PHY_UNKNOWN as the sole value in either of the txPhy or rxPhy parameters
+     *           will result in the default value being used (RxBlePhy.PHY_1M).
```

Comment:
Is a case where `status == GATT_SUCCESS` but the operation was not successful possible? I haven't played with PHY myself.

Reply:
The `status == GATT_SUCCESS` just means that the request was sent successfully. It's ultimately up to the Bluetooth controller to accept the request or deny it. Also, some controllers don't even support PHY updates, while others are optional. For instance, with the nRF52840 you can set it so PHY negotiation isn't an option and you can give it a static PHY you want the controller to run at. I don't believe this would be common for controllers that plan on having mobile app interactions though. It's more for cases where you build two devices, you control both devices, and you want them to only talk to each other. In that case you have control of both ends and can optimize the PHY based on the project. For instance, you have a device at one end and then use CODED PHY to communicate to another device that a mile away type of thing. In that case, PHY 1Mbps and PHY 2Mbps wouldn't work.

Here is part of the documentation:
Set the preferred connection PHY for this app. Please note that this is just a recommendation, whether the PHY change will happen depends on other applications preferences, local and remote controller capabilities. Controller can override these settings. BluetoothGattCallback.onPhyUpdate will be triggered as a result of this call, even if no PHY change happens. It is also triggered when remote device updates the PHY.

To summarize, it's only if the request was sent successfully and nothing more.

Reply:
Yeah, the documentation of BluetoothGatt says that the callback will be called even if nothing changes. In this case I would expect current PHY settings being correctly returned — that would be `PHY_LE_1M` in case the peripheral would be a 4.0 device or other valid result if the peripheral supports it.

In other words I think that `RxBlePhy.PHY_UNKNOWN` value could be useful only if more PHY settings would be added to the Bluetooth Specification in the future. The user has no use-case to use this value on their own. If that's the case maybe we could come up with a better future-proof approach so the library would not need an update as soon as new BT spec is released. WDYT?

We could achieve that by making RxBlePhy an interface that would return a value (and something more probably). That interface could have some static properties that would be its own implementations. Then we would be able to return a Set of predefined values and log warnings when the OS would give us unrecognised values but still return them. WDYT?